prow/plugins/verify-owners/verify-owners.go

```diff
@@ -171,17 +280,38 @@ func handle(ghc githubClient, gc *git.Client, log *logrus.Entry, pre *github.Pul
 		if err != nil {
 			return fmt.Errorf("error creating a review for invalid %s file%s: %v", ownersFileName, s, err)
 		}
-	} else {
-		// Don't bother checking if it has the label...it's a race, and we'll have
-		// to handle failure due to not being labeled anyway.
-		if err := ghc.RemoveLabel(org, repo, pre.Number, labels.InvalidOwners); err != nil {
-			return fmt.Errorf("failed removing %s label: %v", labels.InvalidOwners, err)
+	}
+
+	if len(nonCollaborators) > 0 {
+		if !hasInvalidOwnersLabel {
+			if err := ghc.AddLabel(org, repo, number, labels.InvalidOwners); err != nil {
+				return err
+			}
+		}
+
+		// prune old comments before adding a new one
+		cp.PruneComments(func(comment github.IssueComment) bool {
+			return strings.Contains(comment.Body, fmt.Sprintf(nonCollaboratorResponseFormat, ownersFileName, org))
+		})
+		if err := ghc.CreateComment(org, repo, number, markdownFriendlyComment(org, nonCollaborators)); err != nil {
+			log.WithError(err).Errorf("Could not create comment for listing non-collaborators in %s files", ownersFileName)
 		}
 	}
+
+	// Don't bother checking if it has the label...it's a race, and we'll have
+	// to handle failure due to not being labeled anyway.
+	if err := ghc.RemoveLabel(org, repo, pre.Number, labels.InvalidOwners); err != nil {
```

Comment:
If we added the label in L287 it looks like we'll remove it here??

Reply:
Fixed.